crates/forge/bin/cmd/script/sequence.rs

```diff
@@ -246,6 +246,17 @@ impl ScriptSequence {
         Ok((broadcast, cache))
     }
 
+    /// Checks that there is an Etherscan key for the chain id of this sequence.
+    pub fn verify_preflight_check(&self, config: &Config, verify: &VerifyBundle) -> Result<()> {
+        if config.get_etherscan_api_key(Some(self.chain.into())).is_none() &&
+            verify.verifier.verifier == VerificationProviderType::Etherscan
+        {
+            eyre::bail!("Etherscan API key wasn't found for chain id {}", self.chain)
```

Comment:
I think this error message should mention that deployment is aborted/not started?